credentials_service.py

```
@@ -0,0 +1,174 @@
+            cursor = conn.cursor()
+
+            # Use INSERT OR REPLACE to handle both insert and update
+            cursor.execute('''
+                INSERT OR REPLACE INTO gmail_credentials (email, password, updated_at)
+                VALUES (?, ?, CURRENT_TIMESTAMP)
+            ''', (email, password))
+
+            conn.commit()
+            conn.close()
+            logger.info(f"Successfully stored credentials for: {email}")
+            return True
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)